proto/minder/v1/minder.proto

```diff
@@ -523,33 +551,35 @@ message CheckHealthResponse {
 }
 
 message GetAuthorizationURLRequest {
-    string provider = 1;
-    string project_id = 2;
     bool  cli = 3;
     int32  port = 4;
     optional string owner = 5;
     Context context = 6;
+
+    reserved 1, 2; // deprecated provider and project_id
 }
 
 message GetAuthorizationURLResponse {
     string url = 1;
 }
 
 message ExchangeCodeForTokenCLIRequest {
-    string provider = 1;
-    string project_id = 2;
+    string provider = 1 [deprecated=true];
     string code = 3;
     string state = 4;
     string redirect_uri = 5;
     Context context = 6;
+
+    reserved 2; // deprecated project_id
 }
 
 message StoreProviderTokenRequest {
-    string provider = 1;
-    string project_id = 2;
+    string provider = 1 [deprecated=true];
```

Comment:
This might be my lack of grpc knowledge, but why are some fields marked as deprecated=true and some moved as reserved? If you have a link to a document I could read, I'll be glad to learn more about how to deprecate grpc fields properly

Reply:
The fields that we want to remove but are still used somewhere in the codebase for backwards compatibility reasons are marked with `deprecated=true`. In our case, they are used in the HTTP endpoints, for example `get: "/api/v1/artifacts/{provider}"`. I've added alternative endpoints that don't use the deprecated field, but kept the old ones for backwards compatibility. Marking a field with `deprecated=true` will give a warning when you attempt to use the field, but will not remove the field. [link to docs](https://protobuf.dev/programming-guides/proto3/#options)

The fields that are no longer used in the codebase have their field number marked as reserved to prevent new code from reusing the old field number in the future. [link to docs](https://protobuf.dev/programming-guides/proto3/#fieldreserved)